src/components/table.tsx

```diff
@@ -4,29 +4,45 @@ import { DonationEntry } from '../types/global.d'
 
 interface Props {
     dataset: DonationEntry[],
-    min: number,
-    max: number,
 }
 
 
-const isDisplayed = (value: DonationEntry) => {
+const isDisplayed: (arg0: DonationEntry) => boolean = (value: DonationEntry) => {
     return value.isPublic === true;
 }
 
-const Table: React.FC<Props> = ({ dataset, min, max }) => {
-    let displayData: DonationEntry[];
+//recursive function to avoid writing over variables
+const mapHTML: (displayData: DonationEntry[], listItems: JSX.Element[], index: number) => React.FC | JSX.Element[]
+    = (displayData: DonationEntry[], listItems: JSX.Element[], index: number) => {
+        //create html to add to newlist
+        const newListItem: JSX.Element = <li key={index}>{displayData[index].amount}</li>
 
-    displayData = dataset.filter(isDisplayed);
+        //add html to newList
+        const newListItems: JSX.Element[] = [...listItems, newListItem];
 
-    let tableHTML = dataset.map(function (item, i) {
-        console.log(item)
-        return <li key={i}>{item.amount}</li>
-    })
```

Comment:
Let's return this function back - it's pure and a clearer solution to what's proposed in this PR.